src/content/docs/specs/versioning/get-versions.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: Get supported versions
+description: Query a server for its supported versions
+sidebar:
+  order: 2
+  badge:
+   text: GET
+   variant: success
+---
+
+import CoreAction from "@partials/_core-action.mdx";
+
+<CoreAction />
+
+```http title="Endpoint"
+GET /v1/versions
+```
+
+This endpoint returns an array of versions supported by the server. This informs the client what **core** features are supported. The array MUST contain an entry for each **major** version supported by the server.
+
+:::note
+The `minor_version` MUST reflect the **highest** minor version supported by the server.
+:::
+
+| Field      | Type            | Required? | Description                  |
+| ---------- | --------------- | --------- | ---------------------------- |
+| `versions` | Array\<String\> | Yes       | A list of supported versions |
+
+## Client behavior
+
+Clients SHOULD use the version number to adjust their behavior. For example, if the client is built with support for **v1.4** of the spec, and the server supports up to **v1.2**, the client should modify its calls to ensure they are compatible with **v1.2**.
+
+## Server behavior
+
+Servers MUST respond with all fields, including deprecated fields, to ensure that older clients can interpret the response properly.
+
+## Example request
+
+<Tabs>
+   <TabItem label="JSON">
+
+      ```console
+      $ curl -X 'GET' \
+      '/v1/versions' \
+      -H 'accept: application/json'
+      ```
+
+   </TabItem>
+   <TabItem label="XML">
+
+      ```console
+      $ curl -X 'GET' \
+      '/v1/versions' \
+      -H 'accept: application/xml'
+      ```
+
+   </TabItem>
+</Tabs>
+
+<Tabs>
+   <TabItem label="JSON">
+
+      ```json
+      {
+         "versions": {
+            "v1": {
+               "minor": 6
+            },
+            "v2": {
+               "minor": 0
+            }
+         }
```

Comment:
This should be an array, if I remember correctly

Reply:
```suggestion
         "versions": [
            "v1": {
               "minor": 6
            },
            "v2": {
               "minor": 0
            }
         ]
```

Reply:
@JonOfUs This is actually very hard to achieve in an OpenAPI scheme. Since the key (major version) is an unknown value such as "v1", "v2", "v3", you need to put it in as an additional property. The whole thing is essentially a dictionary where you have a string key and an object value.

Reply:
Would it be easier to also put the major version into the object? Or is it too difficult to read then?

```json
{
  "versions": [
    {
      "major": 1,
      "minor": 6
    },
    {
      "major": 2,
      "minor": 0
    }
  ]
}
```

Or we just stick with "v1.2" and accept the string parsing, then we could also add release candidate versions with e.g. "rc1.6" or something like that for testing.

Reply:
@keunes @georgkrause Any preferences?

Reply:
@Sporiff @JonOfUs I don't think that semantic versioning makes any sense here. We are only talking about the version of the interface, which is basically just the schema for requests and responses for each endpoint. I am failing to consider any change that you wouldn't consider to be breaking, so you would never actually bump a minor version. 

Another discussion is if we need an unstable indicator, but why would it be interesting for the application to know if the API is unstable or not? Its important for third party application developers and therefore should be part of the documentation, not of the API itself.

Reply:
@georgkrause I think you're probably right. For versions, we probably only need to mention the major version support. We should rely on a capabilities endpoint to communicate what capabilities are implemented by the server, and the major version should only communicate changes to core endpoints.

Reply:
> I am failing to consider any change that you wouldn't consider to be breaking, so you would never actually bump a minor version.

@georgkrause I think we foresaw the minor versions to be used for _extended_ functionality (new, optional endpoints or fields). If we want, say, add the 'is_in_queue' field on the `episodes` endpoint, is it worth a new major version?

I think for us, semantically speaking, we could/should still have minor releases, and use git tags (specifically for the Open API specs file).

That said, I guess that this doesn't mean we also need to support minor versions in the version endpoint, right?

> major version should only communicate changes to core endpoints

@Sporiff I would say we'll also need a new (major) version if there are breaking changes in optional endpoints. Imagine we would change the [`download_status`](https://deploy-preview-95--openpodcastapi.netlify.app/specs/episodes/get-all/) from boolean to ENUM without increasing the version, that wouldn't be clear from the capabilities endpoint and could break apps' functionality, no?

Reply:
> @Sporiff I would say we'll also need a new version if there are breaking changes in optional endpoints. Imagine we would change the [download_status](https://deploy-preview-95--openpodcastapi.netlify.app/specs/episodes/get-all/) field from boolean to ENUM without increasing the version, that wouldn't be clear from the capabilities endpoint and could break apps' functionality, no? (And if so, wouldn't minor versions be a better fit for this?)

This is true and we probably need to start a discussion about optional endpoints. Maybe its not a good idea to think about required/optional endpoints. Maybe we should instead have a core API with versioning and then we can allow extensions to the API, that following their own versioning?

Reply:
Those are some interesting points, less of a quick decision than I thought it might be 😋 Maybe that could be a nice topic for Thursday, if you're able to join a meeting then?